### ISSUE TITLE
Clone repository using SSH from examples don't work 

### ISSUE THREAD

**Issue by YaroslavGladun** [libgit2/libgit2#6292]
**Reproduction steps:**
1. I cloned libgit2 and set USE_SSH and BUILD_EXAMPLES as ON:
```
option(USE_SSH                 "Link with libssh2 to enable SSH support"               ON)
option(BUILD_EXAMPLES          "Build library usage example apps"                      ON)
```
2. Then I built it:
```
yaroslav@yaroslav-PC:~/Desktop/libgit2$ mkdir build
yaroslav@yaroslav-PC:~/Desktop/libgit2$ cd build/
yaroslav@yaroslav-PC:~/Desktop/libgit2/build$ cmake ..
yaroslav@yaroslav-PC:~/Desktop/libgit2/build$ cmake --build .
```
3. I have only one SSH key without password (id_rsa) and added it to azure devops. So, I can clone repository using git and SSH URI:
```
yaroslav@yaroslav-PC:~/Desktop$ git clone git@ssh.dev.azure.com:v3/TestOrganization/TestProject/TestRepo
Cloning into 'TestRepo'...
remote: Azure Repos
remote: Found 15 objects to send. (2 ms)
Receiving objects: 100% (15/15), 3.66 MiB | 3.94 MiB/s, done.
Resolving deltas: 100% (2/2), done.
```
4. Then I run example and tried to clone repository:
```
yaroslav@yaroslav-PC:~/Desktop/libgit2/build/examples$ ./lg2 clone git@ssh.dev.azure.com:v3/TestOrganization/TestProject/TestRepo /home/yaroslav/Desktop/TestExamples
SSH Key: /home/yaroslav/.ssh/id_rsa
Password: 
SSH Key: /home/yaroslav/.ssh/id_rsa
Password: 
SSH Key: /home/yaroslav/.ssh/id_rsa
Password: 
SSH Key: /home/yaroslav/.ssh/id_rsa
Password: 
SSH Key: /home/yaroslav/.ssh/id_rsa
Password: 
...
```

**Expected behavior:**
I suppose it should ask path to SSH key and password one time.

**Actual behavior:**
It goes into infinity loop of asking SSH key path and password.

**Version of libgit2 (release number or SHA1):** _bdab22384cc61d315005a65456a9f9563bb27c8f_

**Operating system tested:** _Ubuntu 20.04_

**Remark:**
I tried to use `git_cred_ssh_key_new` in my project but it goes into infinity loop of credential callback. Unfortunately I cannot use ssh agent and need just set path to SSH key.

**Comment by neithernut** [comment]
This looks like a duplicate of #6181.
TL;DR: Ubuntu 20.04 still ships libssh2 1.8.0, which only supports "outdated" ciphers.

**Comment by carlosmn** [comment]
Yeah this looks like a duplicate and the libssh2 version does not support the ciphers that the server requires.
